### PR TITLE
Homework 3 completed

### PR DESCRIPTION
**Pull request by AleksandrDark**

### CODE REVIEW

File: src/test/java/com/epam/tc/hw2/WebDriverElement.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.epam.tc.hw2;
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw2/WebDriverElement.java]: Files from hw2 should not be in PR fro hw3

File: src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package com.epam.tc.hw3.pages;
+
+import java.util.List;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+
+public class DifferentElementsPage extends BasePage {
+    @FindBy(xpath = "//li[@class='menu-title']")
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java]: Let's rewrite this in css form
```suggestion
    @FindBy(css = "li.menu-title")
```

\+ `logs` in the same way

File: src/test/java/com/epam/tc/hw3/pages/LoginPage.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    public void openSite() {
+        webDriver.navigate().to(URL);
+    }
+
+    public void performLogin() {
+        userIcon.click();
+        this.username.sendKeys(property.readProperty("username"));
+        this.password.sendKeys(property.readProperty("password"));
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw3/pages/LoginPage.java]: Pass credentials in method arguments

File: src/test/java/com/epam/tc/hw3/testlogic/HomePageTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.assertj.core.api.SoftAssertions;
+import org.openqa.selenium.WebElement;
+import org.testng.annotations.Test;
+
+public class HomePageTest extends BasePage {
+    @Test
+    public void testHomePage() {
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw3/testlogic/HomePageTest.java]: Please pass all expected values as test method arguments (hint: use dataprovider)

File: src/test/java/com/epam/tc/hw3/util/Property.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package com.epam.tc.hw3.util;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+public class Property {
+
+    public String readProperty(String key) {
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw3/util/Property.java]: A good candidate to become the static method 
And class looks like utility-class, so would be better to make it `final` and revoke the possibility to instantiate it

**Review by shoggy — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/tc/hw2/WebDriverElement.java (hunk 1 above)
Comment (AleksandrDark) [reply src/test/java/com/epam/tc/hw2/WebDriverElement.java]: it is ready

File: src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java (hunk 2 above)
Comment (AleksandrDark) [reply src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java]: it is ready

File: src/test/java/com/epam/tc/hw3/testlogic/HomePageTest.java (hunk 4 above)
Comment (AleksandrDark) [reply src/test/java/com/epam/tc/hw3/testlogic/HomePageTest.java]: ready

File: src/test/java/com/epam/tc/hw3/pages/LoginPage.java (hunk 3 above)
Comment (AleksandrDark) [reply src/test/java/com/epam/tc/hw3/pages/LoginPage.java]: done this remark

File: src/test/java/com/epam/tc/hw3/util/Property.java (hunk 5 above)
Comment (AleksandrDark) [reply src/test/java/com/epam/tc/hw3/util/Property.java]: done this remark

**Review by shoggy — APPROVED** [review]

**Pull request merged**